apps/builder/app/routes/rest.domain.remove.ts

```diff
@@ -0,0 +1,32 @@
+import type { ActionArgs } from "@remix-run/node";
+import { zfd } from "zod-form-data";
+import { prisma } from "@webstudio-is/prisma-client";
+import { trpcClient } from "~/services/trpc.server";
+
+const schema = zfd.formData({
+  domain: zfd.text(),
+  userId: zfd.text(),
+});
+
+export const action = async ({ request }: ActionArgs) => {
+  const { domain, userId } = schema.parse(await request.json());
+
+  const domainFound = await prisma.customDomain.findFirst({
+    where: {
+      userId,
+      domain,
+    },
+  });
+  if (domainFound && domainFound.id) {
+    await trpcClient.customDomain.removeCloudflareCustomHostname.mutate({
+      domain,
+    });
+    await prisma.customDomain.delete({ where: { id: domainFound.id } });
```

Comment:
Must be oss implementation at least with throw new Error, but with same types as SaaS